### PR TITLE
Update kalman sample

### LINKED ISSUE #20091: samples/python/kalman.py is not constant rotation as documented
The comments and documentation in samples/python/kalman.py do not match the behavior of the program.

This line prints that the speed is constant, but it is clear from running the program that the speed is not constant.

https://github.com/opencv/opencv/blob/a604d44d06841565f03da03d8df3124fca639ca3/samples/python/kalman.py#L4

Because of:

https://github.com/opencv/opencv/blob/a604d44d06841565f03da03d8df3124fca639ca3/samples/python/kalman.py#L41
and
https://github.com/opencv/opencv/blob/a604d44d06841565f03da03d8df3124fca639ca3/samples/python/kalman.py#L85-L86

The state variable holds the angle and speed, which at each step is modified by the process transition matrix plus the process noise.  Since the process noise is added onto the state variable for the speed of rotation, the speed is not constant -- it is a random walk.

The random walk in speed (and location) is more visually interesting than tracking a constant rotation, but it isn't required by a Kalman filter and doesn't match the documentation.

Kalman filters can be a confusing topic, and the inconsistent documentation doesn't help.

I suggest either making the process noise zero to match the documentation or replacing the "Rotation speed is constant. " with something like:

Rotation speed is a random walk from N(0,0.1 ), integrating N(0,1e-5) process noise.

If you run the program it is clear that the rotation is not constant. 

Another couple issues:
   * Why plot the crosses underneath the lines?  The thick lines overwrite the points and cover them up.  
   * Why not use/plot the kalman.statePost?  The utility of a Kalman filter is to improve the estimate of the current state based on the recent measurement and to improve the estimate of the next state.  This demo doesn't use the corrected measurement.

### PR DESCRIPTION
**Pull request by AleksandrPanov**
Fixes #20091  

- updated view
- fixed dims in Python sample
- added statePost use

img

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request
- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/ml/misc/java/test/MLTest.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public void testSaveLoad() {
         String filename = OpenCVTestRunner.getTempFileName("yml");
         saved.save(filename);
         SVM loaded = SVM.load(filename);
-        assertTrue(saved.isTrained());
+        assertTrue(loaded.isTrained());
```
Comment (asmorkalov) [inline modules/ml/misc/java/test/MLTest.java:39]: it's not related ti the sample. Please rebase the branch on top of current 3.4.

File: samples/cpp/kalman.cpp [hunk 2]
```diff
@@ -39,7 +43,9 @@ int main(int, char**)
 
     for(;;)
     {
-        randn( state, Scalar::all(0), Scalar::all(0.1) );
+        img = Scalar::all(0);
+        state.at<float>(0) = 0.0f;
+        state.at<float>(1) = 2.f * 3.14159f / 6;
```
Comment (asmorkalov) [inline samples/cpp/kalman.cpp:48]: I recommend to use CV_PI or PI constant.

File: modules/ml/misc/java/test/MLTest.java (hunk 1 above)
Comment (AleksandrPanov) [reply modules/ml/misc/java/test/MLTest.java]: fixed

File: samples/cpp/kalman.cpp (hunk 2 above)
Comment (AleksandrPanov) [reply samples/cpp/kalman.cpp]: fixed using `CV_PI`

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**